custom_components/adaptive_lighting/switch.py

```diff
@@ -1166,21 +1166,26 @@ async def _adapt_light(
         elif supports_colors and adapt_color:
             _LOGGER.debug("%s: Setting rgb_color of light %s", self._name, light)
             service_data[ATTR_RGB_COLOR] = self._settings["rgb_color"]
-
-        context = context or self.create_context("adapt_lights")
-
-        # See #80. Doesn't check if transitions differ but it does the job.
-        last_service_data = self.turn_on_off_listener.last_service_data
-        if not force and last_service_data.get(light) == service_data:
+        # Check if service data differs from the last. See #80.
+        listener = self.turn_on_off_listener
+        last_service_data = listener.last_service_data.get(light)
+        ignore_fields = {ATTR_TRANSITION}
```

Comment:
Could you explain to me why we need to do this?

Reply:
Transitions can easily differ between the last adapt. i.e. initial vs normal transition vs sleep transition vs apply() vs change_switch_settings.

I feel if we're going to check the last service data we might as well check the relevant portions of it. The `force` variable should be the only thing that would cause an adapt if the service data is the same.